igvc_platform/src/tests/test_differential_drive.cpp

```diff
@@ -0,0 +1,202 @@
+//
+// Created by vivek on 3/11/20.
+//
+
+#include <gtest/gtest.h>
+#include <igvc_msgs/velocity_pair.h>
+#include <ros/ros.h>
+#include <geometry_msgs/Twist.h>
+#include <mocking_utils/mock_subscriber.h>
+
+class TestDifferentialDrive : public testing::Test
+{
+public:
+  TestDifferentialDrive() : mock_joy_pub(handle.advertise<geometry_msgs::Twist>("/cmd_vel", 1))
+  {
+  }
+
+protected:
+  [[nodiscard]] bool waitForSubscriber() const {
+    const double timeout = 5.0;
+    const double sleep_time = 1.0;
+    ros::Time end = ros::Time::now() + ros::Duration(timeout);
+    while (mock_joy_pub.getNumSubscribers() == 0)
+    {
+      ros::Duration(sleep_time).sleep();
+      if (ros::Time::now() > end)
+      {
+        return false;
+      }
+    }
+    return true;
```

Comment:
I think this method should belong in the `mocking_utils` somewhere, as I think almost all the tests will want to use this method